pkgs/top-level/linux-kernels.nix

```diff
@@ -290,31 +290,116 @@ in
+              release = "25.11";
+              reason = "it has reached its end of life upstream";
+            };
+
+            linux_5_10_hardened = {
+              type = "alias";
+              release = "25.11";
+              reason = "linux_hardened on nixpkgs only contains latest stable and latest LTS";
+            };
+            linux_5_15_hardened = {
+              type = "alias";
+              release = "25.11";
+              reason = "linux_hardened on nixpkgs only contains latest stable and latest LTS";
+            };
+            linux_6_1_hardened = {
+              type = "alias";
+              release = "25.11";
+              reason = "linux_hardened on nixpkgs only contains latest stable and latest LTS";
+            };
+            linux_6_6_hardened = {
+              type = "alias";
+              release = "25.11";
+              reason = "linux_hardened on nixpkgs only contains latest stable and latest LTS";
+            };
+
+            linux_4_19_hardened = {
+              type = "alias";
+              release = "25.11";
+              reason = "linux 4.19 was removed because it will reach its end of life within 24.11";
+            };
+            linux_5_4_hardened = {
+              type = "alias";
+              release = "25.11";
+              reason = "linux_5_4_hardened was removed because it was broken";
+            };
+            linux_6_9_hardened = {
+              type = "alias";
+              release = "25.11";
+              reason = "linux 6.9 was removed because it has reached its end of life upstream";
+            };
+            linux_6_10_hardened = {
+              type = "alias";
+              release = "25.11";
+              reason = "linux 6.10 was removed because it has reached its end of life upstream";
+            };
+            linux_6_11_hardened = {
+              type = "alias";
+              release = "25.11";
+              reason = "linux 6.11 was removed because it has reached its end of life upstream";
+            };
+            linux_6_13_hardened = {
+              type = "alias";
+              release = "25.11";
+              reason = "linux 6.13 was removed because it has reached its end of life upstream";
+            };
+            linux_6_14_hardened = {
+              type = "alias";
+              release = "25.11";
+              reason = "linux 6.14 was removed because it has reached its end of life upstream";
+            };
+            linux_6_15_hardened = {
+              type = "alias";
+              release = "25.11";
+              reason = "linux 6.15 was removed because it has reached its end of life upstream";
+            };
+
+            linux_ham = {
+              type = "alias";
+              release = "25.11";
+              reason = "in favour of the standard kernel packages";
+            };
+          };
+        in
+        (lib.mapAttrs (lib.makeAlias kernels) aliases) // { _isAlias = name: aliases ? ${name}; }
```

Comment:
Yuck. This is too much ceremony – I think we have to find something better here. Whether that’s just defining these under an `__aliases` sub‐attribute‐set, or having these in line as `mkAlias` calls adjacent to the regular packages, or whatever. (I haven’t reviewed the implementation strategy enough to predict the issues these will cause, but I think this is no good, and I believe we can do better by making our scope creation functions etc. smarter.)